File: src/main/java/com/gmail/borlandlp/minigamesdtools/geoip/ipstack/IpstackResponseParser.java
package com.gmail.borlandlp.minigamesdtools.geoip.ipstack;

import com.gmail.borlandlp.minigamesdtools.geoip.GeoData;
import com.gmail.borlandlp.minigamesdtools.geoip.LangData;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;

public class IpstackResponseParser {
    private JSONObject jsonObject;

    public IpstackResponseParser(JSONObject json) {
        this.jsonObject = json;
    }

    public GeoData parse() {
        GeoData geoData = new GeoData();

        geoData.ip = (String) jsonObject.get("ip");
        geoData.type = (String) jsonObject.get("type");
        geoData.continentCode = (String) jsonObject.get("continent_code");
        geoData.continentName = (String) jsonObject.get("continent_name");
        geoData.countryCode = (String) jsonObject.get("country_code");
        geoData.continentName = (String) jsonObject.get("country_name");

        JSONObject locSection = (JSONObject) jsonObject.get("location");
        geoData.locGeonameId = (String) locSection.get("geoname_id");
        geoData.locCapital = (String) locSection.get("capital");

        JSONArray langSection = (JSONArray) (locSection).get("languages");
        JSONObject langObj = null;
        for (int i = 0; i < langSection.size(); i++) {
            langObj = (JSONObject) langSection.get(i);
            LangData langData = new LangData();

            langData.langCode = (String) langObj.get("code");
            langData.langName = (String) langObj.get("name");
            langData.langNativeCode = (String) langObj.get("native");

            geoData.languages.add(langData);
        }

        return geoData;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/geoip/ipstack/IpstackResponseParser.kt
package com.gmail.borlandlp.minigamesdtools.geoip.ipstack

import com.gmail.borlandlp.minigamesdtools.geoip.GeoData
import com.gmail.borlandlp.minigamesdtools.geoip.LangData
import org.json.simple.JSONArray
import org.json.simple.JSONObject

class IpstackResponseParser(private val jsonObject: JSONObject) {
    fun parse(): GeoData {
        val geoData = GeoData
        geoData.ip = jsonObject["ip"] as String?
        geoData.type = jsonObject["type"] as String?
        geoData.continentCode = jsonObject["continent_code"] as String?
        geoData.continentName = jsonObject["continent_name"] as String?
        geoData.countryCode = jsonObject["country_code"] as String?
        geoData.continentName = jsonObject["country_name"] as String?
        val locSection = jsonObject["location"] as JSONObject?
        geoData.locGeonameId = locSection!!["geoname_id"] as String?
        geoData.locCapital = locSection["capital"] as String?
        val langSection = locSection["languages"] as JSONArray?
        var langObj: JSONObject?
        for (i in langSection!!.indices) {
            langObj = langSection[i] as JSONObject?
            val langData = LangData
            langData.langCode = langObj!!["code"] as String?
            langData.langName = langObj["name"] as String?
            langData.langNativeCode = langObj["native"] as String?
            geoData.languages.add(langData)
        }
        return geoData
    }
}